msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ServiceBundle.java

```diff
@@ -10,13 +10,15 @@ class ServiceBundle {
     private ExecutorService executorService;
     private TelemetryManager telemetryManager;
     private IHttpClient httpClient;
+    private IHttpHelper httpHelper;
     private ServerSideTelemetry serverSideTelemetry;
 
-    ServiceBundle(ExecutorService executorService, IHttpClient httpClient,
-                  TelemetryManager telemetryManager) {
+    ServiceBundle(ExecutorService executorService, TelemetryManager telemetryManager,
+                  IHttpClient httpClient, IHttpHelper httpHelper) {
```

Comment:
Do you still need the httpClient in the bundle if you have the higher level construct in there? (the helper) ?

Reply:
It turns out the the HttpClient set in the ServiceBundle was only used in one place, to pass an instance of it into HttpHelper. I assume it was because (before this PR) all of HttpHelper's methods were used statically, so HttpHelper wasn't really intended to have a constructor where an HttpClient could be set.

However, since this PR made it all non-static, in the latest commit I added HttpClient to a new constructor in HttpHelper, and removed it from ServiceBundle's constructor.